rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,351 @@
+                    "HTTPConduit was null for " + this + ". Unable to configure security.");
+        }
+
+        if (StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {
+            String asciiString = cxfClient.getBaseURI().toASCIIString();
+            if (!StringUtils.startsWithIgnoreCase(asciiString, "https")) {
+                throw new SecurityServiceException(
+                        "Cannot secure non-https connection " + asciiString);
+            }
+            if (httpConduit.getAuthorization() != null) {
+                httpConduit.getAuthorization().setUserName(username);
+                httpConduit.getAuthorization().setPassword(password);
+            }
+        }
+
+        TLSClientParameters tlsParams = getSecuritySettingsService().getTLSParameters();
+        httpConduit.setTlsClientParameters(tlsParams);
+    }
+
+    public class SecureCxfClientFactory<T> {
+
+        private Client cxfClient;
+
+        private SecureCxfClientFactory(Client cxfClient) {
+            this.cxfClient = cxfClient;
+        }
+
+        /**
+         * Clients produced by this method will be secured with basic authentication
+         * (if a username and password were provided), two-way ssl,
+         * and the provided security subject.
+         * <p/>
+         * The returned client should NOT be reused between requests!
+         * This method should be called for each new request in order to ensure
+         * that the security subject is up-to-date each time.
+         *
+         * @see SecureCxfClientFactoryBuilder
+         */
+        public T getClientForSubject(Subject subject) throws SecurityServiceException {
+            String asciiString = cxfClient.getBaseURI().toASCIIString();
+            if (!StringUtils.startsWithIgnoreCase(asciiString, "https")) {
+                throw new SecurityServiceException(
+                        "Cannot secure non-https connection " + asciiString);
+            }
+
+            WebClient newClient = WebClient.fromClient(cxfClient);
+
+            if (subject instanceof ddf.security.Subject) {
+                RestSecurity.setSubjectOnClient((ddf.security.Subject) subject, newClient);
+            } else {
+                throw new SecurityServiceException("Not a ddf subject " + subject);
+            }
+            WebClient.getConfig(newClient).getRequestContext()
+                    .put(org.apache.cxf.message.Message.MAINTAIN_SESSION, Boolean.TRUE);
+
+            return (T) newClient;
+        }
+
+        /**
+         * Clients produced by this method will be secured with basic authentication
+         * (if a username and password were provided), two-way ssl,
+         * and the system security token (x509 cert).
+         */
+        public T getClientForSystem() throws SecurityServiceException {
+            String asciiString = cxfClient.getBaseURI().toASCIIString();
+            if (!StringUtils.startsWithIgnoreCase(asciiString, "https")) {
```

Comment:
Insecure in this context means HTTP only, with no security or authentication information provided. On second thought, I question the value of that functionality in this wrapper, since without https there's really no benefit to be gained from using the wrapper.